### PR TITLE
Use ostream << operator to get to_string

### PR DESCRIPTION
**Pull request by reyoung**
* Make `PADDLE_ENFORCE_EQ` supports custom class, like DDim

### CODE REVIEW

File: paddle/string/to_string_test.cc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  using namespace paddle::string;
+  ASSERT_EQ(std::to_string(10), to_string(10));
+  ASSERT_EQ("abc", to_string("abc"));
+
+  auto std_to_string = std::to_string(1.2);
+  auto my_to_string = to_string(1.2);
+
+  // std::to_string might fill zero after float value, like 1.2000
+  for (size_t i = 0; i < my_to_string.size(); ++i) {
+    ASSERT_EQ(my_to_string[i], std_to_string[i]);
```
Comment (qingqing01) [inline paddle/string/to_string_test.cc]: 可以去掉std::to_string，直接改成"10", "1.2"来比较吗？ 而不是加`#if !defined(ANDROID) && !defined(__ANDROID__)` 来避免

Comment (reyoung) [inline paddle/string/to_string_test.cc]: 其实也行。。那我这么干吧。

Comment (reyoung) [inline paddle/string/to_string_test.cc]: Done.

File: paddle/string/to_string_test.cc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include "paddle/string/to_string.h"
+#include <gtest/gtest.h>
+
+constexpr char OUT_STR[] = "User Defined Output";
```
Comment (qingqing01) [inline paddle/string/to_string_test.cc]: https://google.github.io/styleguide/cppguide.html#Constant_Names

Variables declared **constexpr** or const, and whose value is fixed for the duration of the program, are named with a leading "k" followed by mixed case. For example:

```
const int kDaysInAWeek = 7;
```

Comment (reyoung) [inline paddle/string/to_string_test.cc]: Done.

**Pull request merged**